### PR TITLE
fix(profile): hide release countdown after release date passes (JOV-1990)

### PR DESCRIPTION
**Pull request by itstimwhite**
## Summary

Fixes [JOV-1990](https://linear.app/jovie/issue/JOV-1990) — the release countdown timer was still rendering on already-released items.

ISR-cached profile pages (`/[username]`, ~1h revalidate) could render the release countdown after the release had actually dropped, because the page snapshot evaluated `releaseDate > now` at cache time. On revisit, the cached HTML still showed "Drops in …" instead of "Out Now / Listen Now".

## Changes

- New `useReleaseAwareNow(releaseDate)` hook (`apps/web/hooks/useReleaseAwareNow.ts`): returns a `Date` that re-renders the calling component exactly once when the release boundary passes. No-op when the date is null/invalid/past.
- `LatestReleaseCard`, `ProfileHomeRail`, `ProfilePrimaryActionCard`: wire `useReleaseAwareNow` into the `isCountdown` / `isFutureRelease` evaluation so the chrome transitions live without a manual refresh, even on a stale ISR snapshot.
- `ReleaseCountdown`: add a defensive synchronous client-side expiry check (`releaseDate <= Date.now()` after hydration) so the timer never renders against a past releaseDate, even if the parent was server-rendered stale.
- New test: `apps/web/tests/unit/hooks/useReleaseAwareNow.test.ts` (5 tests) — locks in the boundary re-render behavior with fake timers.

## Test plan

- [x] `pnpm --filter @jovie/web run typecheck -- --pretty false`
- [x] `pnpm biome check --write apps/web/...` (no fixes applied)
- [x] `vitest run` for `release-visibility.test.ts`, `release-countdown.test.tsx`, `useReleaseAwareNow.test.ts` — 34/34 pass
- [ ] Manual browser check: load a profile with `releaseDate` set to ~30s in the future, observe transition from "Drops in" → "Out Now" without manual refresh

## Summary by CodeRabbit

* **New Features**
  * Added release-boundary–aware timing (new client hook) so profile cards, primary action cards, and countdowns update precisely at release moments.

* **Bug Fixes**
  * Resolved stale cached pages not flipping from "Coming Soon" to "Out Now".
  * Countdown now defensively treats invalid/past release timestamps and avoids rendering when already expired.

* **Tests**
  * Added unit tests covering release-aware timing across date inputs and edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/JovieInc/Jovie/pull/8539)

### CODE REVIEW

**Comment by itstimwhite** [comment]
## Review Summary
**Verdict**: Approve

The PR correctly addresses the stale ISR issue by introducing a client-side hook that re-renders components exactly when a release date passes. The implementation is clean, well-tested, and follows Jovie conventions.

### Critical Issues
None.

### Warnings
- **apps/web/hooks/useReleaseAwareNow.ts:18-19** — The hook unconditionally calls `setNow(new Date())` when `releaseDate` is already in the past. Since the initial state is already `new Date()` at mount, this triggers an unnecessary re-render. Consider guarding with a check like `if (target.getTime() <= Date.now() && now.getTime() <= target.getTime())` to avoid the extra render. Not a bug, but a minor performance concern.

- **apps/web/hooks/useReleaseAwareNow.ts:28** — The 50ms buffer (`msUntil + 50`) is safe but could theoretically cause the hook to miss the exact boundary if the system clock jumps backward. Consider using `Math.max(0, msUntil)` with a small positive value (e.g., 0) to avoid any delay. The current approach is acceptable for practical purposes.

### Suggestions
- **apps/web/hooks/useReleaseAwareNow.ts** — Consider memoizing the `releaseDate` comparison to avoid re-scheduling the timeout on every render when the prop reference changes but the value is the same. This could be done by storing the target timestamp in a ref and comparing. Not required for correctness, but would reduce unnecessary effect runs.

- **apps/web/tests/unit/hooks/useReleaseAwareNow.test.ts** — Add a test case for a past date string (e.g., `'2026-05-10T00:00:00Z'`) to ensure the hook does not schedule a timeout and returns a date >= now. Currently only past `Date` objects are tested.

- **apps/web/components/features/release/ReleaseCountdown.tsx:95** — The `isPastNow` check uses `releaseDate.getTime() <= Date.now()`. Consider using `<` instead of `<=` to avoid hiding the countdown at the exact millisecond of release (though the interval would clear it anyway). This is cosmetic.

### What Looks Good
- The `useReleaseAwareNow` hook is well-documented and handles all edge cases (null, invalid, past, future dates).
- The defensive check in `ReleaseCountdown` ensures stale server-rendered countdowns are immediately hidden on the client.
- The hook is integrated cleanly into three profile components without breaking existing props or logic.
- Tests cover the main scenarios with fake timers, including string dates and invalid inputs.
- No emoji, no native dialogs, no decorative motion — all Jovie conventions respected.
- Conventional commit message format used.

**Review by itstimwhite — COMMENTED** [review]
Automated code review from Hermes PR Review Bot (deepseek/deepseek-v4-flash).

File: apps/web/hooks/useReleaseAwareNow.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    const msUntil = target.getTime() - Date.now();
+    if (msUntil <= 0) {
+      setNow(new Date());
+      return;
+    }
+
+    const timeout = setTimeout(() => setNow(new Date()), msUntil + 50);
+    return () => clearTimeout(timeout);
+  }, [releaseDate]);
```
[the remark this answers is not in the document]
Comment (itstimwhite) [reply apps/web/hooks/useReleaseAwareNow.ts]: Fixed in commit 1710e340f. The effect had two issues: (1) setNow was called for past dates, causing a spurious re-render — now bails early with plain return; (2) replaced raw releaseDate dep with stable releaseTimestamp number so callers that construct new Date() each render don't trigger unnecessary effect re-runs.

File: apps/web/components/features/release/ReleaseCountdown.tsx [hunk 2]
```diff
@@ -89,8 +89,14 @@ export function ReleaseCountdown({
...
-  if (!timeLeft || timeLeft.total <= 0) {
+  // Defensive synchronous expiry check: if an ISR-cached parent rendered us
+  // with a release date already in the past, render nothing immediately
+  // instead of waiting for the next interval tick. The `typeof window` guard
+  // preserves SSR-safe hydration (server matches the initial null state).
+  const isPastNow =
+    typeof window !== 'undefined' && releaseDate.getTime() <= Date.now();
+
+  if (!timeLeft || timeLeft.total <= 0 || isPastNow) {
     return null;
```
[the remark this answers is not in the document]
Comment (itstimwhite) [reply apps/web/components/features/release/ReleaseCountdown.tsx:108]: Fixed in commit 1710e340f. Added a `Number.isFinite(targetMs)` guard at the top of `getTimeLeft()` that returns the zero-state `{ days: 0, hours: 0, minutes: 0, total: 0 }` for any non-finite timestamp (NaN from `new Date('invalid').getTime()` or ±Infinity). This ensures the component renders `null` immediately rather than propagating NaN into countdown digits.

File: apps/web/tests/unit/hooks/useReleaseAwareNow.test.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+  it('does not schedule a re-render when releaseDate is already in the past', () => {
+    const past = new Date('2026-05-10T00:00:00Z');
+    const { result } = renderHook(() => useReleaseAwareNow(past));
+    const initial = result.current;
+    act(() => {
+      vi.advanceTimersByTime(60_000);
+    });
+    expect(result.current).toBe(initial);
+  });
```
[the remark this answers is not in the document]
Comment (itstimwhite) [reply apps/web/tests/unit/hooks/useReleaseAwareNow.test.ts:28]: Addressed in commit 1710e340f. Added the rerender regression test as suggested — it creates two `new Date(pastIso)` instances with the same underlying timestamp, rerenders, advances timers, and asserts strict object identity is preserved. This guards against the update-loop regression class caused by dependency-identity churn.

File: apps/web/hooks/useReleaseAwareNow.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+  useEffect(() => {
+    // Bail out for null/invalid or already-past dates — no timer needed.
+    if (!Number.isFinite(releaseTimestamp)) return;
+    const msUntil = releaseTimestamp - Date.now();
+    if (msUntil <= 0) return;
+
+    const timeout = setTimeout(() => setNow(new Date()), msUntil + 50);
+    return () => clearTimeout(timeout);
+  }, [releaseTimestamp]);
```
[the remark this answers is not in the document]
Comment (itstimwhite) [reply apps/web/hooks/useReleaseAwareNow.ts]: Fixed in commit 816e2f612. Implemented the chunked scheduling approach: bail early for already-past dates (no re-render scheduled), then for future dates use `Math.min(msUntil + 50, MAX_TIMEOUT_MS)` to schedule the first tick. If the timer fires and the boundary still hasn't passed, `schedule()` recursively reschedules at `Math.min(remaining, MAX_TIMEOUT_MS)`. All 6 unit tests pass including the past-date no-re-render contract.
